support more architectures

diceroll123 opened the pull request:
## Summary

Support a few more architectures

## Test Plan

I have not tested it. Ideally the PR will do a dry run.

diceroll123 commented:
Guess not!

konstin commented:
We have to manually approve CI runs each times for first-time contributors unfortunately. You can activate github actions in your fork and make a PR in your fork to test.

diceroll123 commented:
Updated.

[In my PR on my fork](https://github.com/diceroll123/ruff-vscode/pull/1), I:
- made it so release workflow runs on PR, for testing purposes
- disabled macos builds because they cost money apparently
- removed the build ID and publish steps because I didn't need them

But the only changes I've brought over to here are the build architectures. Attached is the list of successful actions!

<img width="291" alt="image" src="https://github.com/astral-sh/ruff-vscode/assets/1243957/ff180173-4ec3-43f7-b3ff-78923d0edac3">

`aarch64-unknown-linux-musl` I believe is what will allow us to use ruff within a devcontainer running on alpine linux, can't wait! 😄

charliermarsh commented:
I can do some testing and ultimately build for nightly, thanks :)

charliermarsh commented on .github/workflows/release.yaml:
```diff
@@ -68,6 +68,17 @@ jobs:
+          - os: ubuntu-20.04
+            target: armv7-unknown-linux-musleabihf
+            code-target: linux-armhf
+            arch: armv7
```
I think this isn't quite right because these `code-target` values are the same as the ones above. We need to support `alpine-x64` and `alpine-arm64`. See: https://code.visualstudio.com/api/working-with-extensions/publishing-extension#platformspecific-extensions.

charliermarsh replied:
I need to play around with things (https://github.com/astral-sh/ruff-vscode/pull/387) to understand how this works. I think we need the extension itself to be built under Alpine.

charliermarsh replied:
@diceroll123 -- Okay, I think I got this working [here](https://github.com/astral-sh/ruff-vscode/pull/387/files) -- notice how the build is done within an Alpine container, does that make sense?

diceroll123 replied:
Absolutely! Awesome stuff. 😄 Learning a lot from ya.

charliermarsh replied:
Thanks for doing this, and sorry that it didn't get merged!
